### PR TITLE
Revert "Revert "switch country authorization from groups to nrc organ…

### PR DESCRIPTION
**Pull request by internetti**
…isation""

### CODE REVIEW

**Comment by neb42** [comment]
conflicts

File: internal/server/middleware/auth.go [hunk 1]
```diff
@@ -158,6 +160,9 @@ func validateTokenClaims(claims TokenClaims) error {
 	if claims.Groups == nil {
 		return fmt.Errorf("token is missing groups")
 	}
+	if claims.NrcOrganisation == "" {
```
Comment (ludydoo) [inline internal/server/middleware/auth.go:163]: This should be removed

File: internal/server/middleware/auth.go [hunk 2]
```diff
@@ -15,12 +15,13 @@ import (
...
-	Iss    string   `json:"iss"`
-	Email  string   `json:"email"`
-	Groups []string `json:"groups"`
-	Iat    int64    `json:"iat"`
-	Exp    int64    `json:"exp"`
+	Sub             string   `json:"sub"`
+	Iss             string   `json:"iss"`
+	Email           string   `json:"email"`
+	Groups          []string `json:"groups"`
+	NrcOrganisation string   `json:"nrcOrganisation"`
```
Comment (ludydoo) [inline internal/server/middleware/auth.go:22]: `NrcOrganisation` should be removed

File: internal/server/middleware/auth_test.go [hunk 3]
```diff
@@ -13,50 +13,66 @@ func Test_validateTokenClaims(t *testing.T) {
...
+				Sub:             "subject",
+				Email:           "email",
+				NrcOrganisation: "NRC Uganda",
+				Exp:             1,
+				Iat:             2,
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing nrcOrganisation",
```
Comment (ludydoo) [inline internal/server/middleware/auth_test.go:75]: This test case should be removed

File: internal/db/migrations/postgres/018_add_nrc_organisation_to_country.up.sql [hunk 4]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE countries
+    DROP COLUMN jwt_group;
+ALTER TABLE countries
+    -- add nrc_organisation column
+    ADD COLUMN nrc_organisation      VARCHAR(255)         NOT NULL DEFAULT '';
+
```
Comment (neb42) [inline internal/db/migrations/postgres/018_add_nrc_organisation_to_country.up.sql:6]: `ALTER TABLE countries RENAME COLUMN jwt_group TO nrc_organisation;` - will help move over all the values in dev

**Review by neb42 — APPROVED** [review]

**Pull request merged**